src/components/ExpandedNode/ExpandedNode.tsx

```diff
@@ -81,11 +88,22 @@ const ExpandedNode: FC<ExpandedNodeProps> = memo(
       setShowReport(false);
     };
 
+    const onAdvance = (): void => {
+      const newPatientRecords = [...patientRecords];
+      const content = `${pathwayState.label} - Advance`;
+      const documentReference = createDocumentReference(content, patient);
+
+      newPatientRecords.push(documentReference);
+      client?.create?.(documentReference);
+      setPatientRecords(newPatientRecords);
```

Comment:
In other places I think we use this to be more concise
```suggestion
      setPatientRecords([...patientRecords, documentReference]);
```